src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -340,13 +310,54 @@ const SafetyStatus SafetyStatusGet(chip::EndpointId endpoint)
     return status;
 }
 
+LimitStatus CheckLimitState(uint16_t position, AbsoluteLimits limits)
+{
+
+    if (limits.open > limits.closed)
+        return LimitStatus::Inverted;
+
+    if (position == limits.open)
+        return LimitStatus::IsUpOrOpen;
+
+    if (position == limits.closed)
+        return LimitStatus::IsDownOrClose;
+
+    if ((limits.open > 0) && (position < limits.open))
+        return LimitStatus::IsOverUpOrOpen;
```

Comment:
IsPastUpOrOpen might be clearer, since "Over" for something that is smaller is odd.